src/scenario/custom_messages.c

```diff
@@ -0,0 +1,325 @@
+#include "custom_messages.h"
+
+#include "core/array.h"
+#include "core/encoding.h"
+#include "core/file.h"
+#include "core/log.h"
+#include "core/string.h"
+
+#define CUSTOM_MESSAGES_ARRAY_SIZE_STEP 100
+
+#define AUDIO_FILE_PATHS_COUNT 4
+static const char AUDIO_FILE_PATHS[][32] = {
+    "community/audio/",
+    "community/briefing_audio/",
+    "mp3/",
+    "wavs/",
+};
+
+#define VIDEO_FILE_PATHS_COUNT 2
+static const char VIDEO_FILE_PATHS[][32] = {
+    "community/video/",
+    "smk/",
+};
+
+static array(custom_message_t) custom_messages;
+
+static struct {
+    char filepath_media[FILE_NAME_MAX];
+    char filepath_audio[FILE_NAME_MAX];
+    char temp_filepath[FILE_NAME_MAX];
+} data;
+
+static int entry_in_use(const custom_message_t *entry)
+{
+    return entry->in_use != 0;
+}
+
+static void new_entry(custom_message_t *obj, int position)
+{
+    obj->id = position;
+}
+
+void custom_messages_clear(void)
+{
+    custom_messages.size = 0;
+    if (!array_init(custom_messages, CUSTOM_MESSAGES_ARRAY_SIZE_STEP, new_entry, entry_in_use)) {
+        log_error("Unable to allocate enough memory for the custom media array. The game will now crash.", 0, 0);
+    }
+}
+
+void custom_messages_clear_all(void)
+{
+    custom_messages_clear();
+    custom_media_clear();
+    message_media_text_blob_clear();
+}
+
+custom_message_t *custom_messages_get(int message_id)
+{
+    return array_item(custom_messages, message_id);
+}
+
+int custom_messages_get_id_by_uid(const uint8_t *message_uid, int *message_id)
```

Comment:
Instead of having a pointer with the id, why not return the id? If the message isn't found, return `0`.

If `0` is used, make it unused. That is standard practice in all other dynamic elements anyway.

Reply:
Done.